packages/channel-utils/src/date-utils.ts

```diff
@@ -1,32 +1,35 @@
 import { addInterval, getDayPriorTillTomorrow, getLastXMonths, getTomorrowStartOfDay } from '@repo/utils';
 import { prisma } from '@repo/database';
+import { tierConstraints } from '@repo/mappings';
 
 export const timeRanges = async (initial: boolean, adAccountId: string): Promise<{ since: Date; until: Date }[]> => {
   if (initial) {
     const range = getLastXMonths();
     return splitTimeRange(range.since, range.until);
   }
   const latestInsight = await prisma.insight.findFirst({
-    select: { date: true },
+    select: { date: true, adAccount: { select: { organizations: { select: { tier: true } } } } },
     where: { adAccountId },
     orderBy: { date: 'desc' },
   });
-  const furthestDate = addInterval(new Date(), 'day', -maxTimePeriodDays);
+
+  const organizationTier = latestInsight?.adAccount?.organizations[0]?.tier;
+
+  const furthestDate = addInterval(new Date(), 'day', -tierConstraints[organizationTier || 'Launch'].maxRecency);
   const range = latestInsight ? getDayPriorTillTomorrow(latestInsight.date) : getDayPriorTillTomorrow(furthestDate);
-  return splitTimeRange(range.since, range.until);
+  return splitTimeRange(range.since, range.until, tierConstraints[organizationTier || 'Launch'].maxRecency);
 };
 
-const maxTimePeriodDays = 90; // 90 days
-const maxTimePeriod = 1000 * 60 * 60 * 24 * maxTimePeriodDays;
-export const splitTimeRange = (since: Date, until: Date = new Date()): { since: Date; until: Date }[] => {
+export const splitTimeRange = (since: Date, until: Date = new Date(), maxTimePeriodDays?: number): { since: Date; until: Date }[] => {
```

Comment:
```suggestion
export const splitTimeRange = (since: Date, until: Date = new Date(), maxTimePeriodDays: number = 7): { since: Date; until: Date }[] => {
```
How about using this, because if we use the tierContraints value rather then hardcoding 7 below in the function, we'll need to pass another optional parameter.... maybe `tier` or `minRecency`.

Reply:
I think it is essential for this functiona to have a tier as parameter yes.